.github/workflows/e2e.yml

```diff
@@ -0,0 +1,57 @@
+name: Turbinia E2E Test
+
+on: [push, pull_request]
+
+jobs:
+  e2e-test:
+    name: Run local stack e2e test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Pull docker images for cache
+      run: |
+        docker pull ubuntu:18.04
+        docker pull us-docker.pkg.dev/osdfir-registry/turbinia/release/turbinia-server-dev:latest
+        docker pull us-docker.pkg.dev/osdfir-registry/turbinia/release/turbinia-worker-dev:latest
+    - name: Build Turbinia server docker image
+      run: docker build --cache-from=ubuntu:18.04,us-docker.pkg.dev/osdfir-registry/turbinia/release/turbinia-server-dev:latest -t turbinia-server-dev -f docker/server/Dockerfile .
+    - name: Build Turbinia server worker image
+      run: docker build --cache-from=ubuntu:18.04,us-docker.pkg.dev/osdfir-registry/turbinia/release/turbinia-worker-dev:latest -t turbinia-worker-dev -f docker/worker/Dockerfile .
+    - name: Prepare Turbinia configuration and e2e evidence
+      run: |
+        mkdir ./conf
+        mkdir ./evidence
+        chmod 777 ./evidence
+        sed -f ./docker/local/local-config.sed ./turbinia/config/turbinia_config_tmpl.py > ./conf/turbinia.conf
+        curl https://raw.githubusercontent.com/obsidianforensics/hindsight/master/tests/fixtures/profiles/60/History > History
+        tar -vzcf ./evidence/history.tgz History
+    - name: Verify Turbinia configuration and evidence
+      run: |
+        cat ./conf/turbinia.conf
+        ls -al ./evidence
+    - name: Startup local turbinia docker-compose stack
+      run: docker-compose -f ./docker/local/docker-compose.yml up -d
+    - name: Sleep for 5s
+      uses: juliangruber/sleep-action@v1
+      with:
+        time: 5s
+    - name: Show running instances
+      run: docker ps -a
+    - name: Show container logs
+      run: |
+        docker logs turbinia-server
+        docker logs turbinia-worker
+    - name: Run E2E test
+      run: |
+        docker exec -t turbinia-server turbiniactl compresseddirectory -l /evidence/history.tgz > request.txt
```

Comment:
Would it be better move the run steps into a script so it's more re-usable and easier to extend later too?  I'm guessing we'll want to add more to this as time goes on.  If this is only intended to be a quick supplement to the other e2e testing infrastructure then it's probably ok, but this format might be a little limiting if not.

Reply:
I want to replace this with 1 call to a disk image containing artifacts that activate each worker. So it will still be 1 line. I don't think we need to make this more difficult by pulling in external scripts. The same one liner I will use for the hybrid and gcp only tests.

Reply:
The sending of the request is fairly straightforward and can be a one-liner, but won't we still want to do things like check the outputs for each of the tasks, etc?  Is this intended as a supplement to the other e2e tests so we have something quick/visible in PRs?  Or do you think this will eventually replace the other scripts?  If the former, I think this is probably fine, but I would prefer that we don't have to use this format for all of our e2e tests.

Reply:
I have refactored the e2e test out of the github action and added it to an external file so we can adjust it easily later. (eg add different types of evidence to it)